src/section/section_keys.rs

```diff
@@ -102,8 +102,10 @@ impl SectionKeysProvider {
         self.current = dkg_result
             .secret_key_share
             .and_then(|secret_key_share| {
-                elders_info
-                    .position(our_name)
+                dkg_key
+                    .0
+                    .iter()
+                    .position(|id| id.name() == our_name)
```

Comment:
This won't always work, because the actual participants might be different from the expected participants (in case of some non-cooperating nodes) and this will then return incorrect index. I have a fix for this in the other PR: https://github.com/maidsafe/routing/pull/2185/commits/d5cf05db1ede665ef3134ecfb7675ab54f8bba5c

Reply:
Yeah, I see your fix, will cherry-pick that into this PR to overwrite mine.